packages/components/data-table/src/data-table.styles.tsx

```diff
@@ -103,6 +104,23 @@ const TableGrid = styled.table<TTableGrid>`
 
 const TableHeader = styled.thead`
   display: contents;
+  /* Adds right border that doesn't affect column width
+  *  to each header cell when table header is hovered
+  */
+  :hover {
+    ${BaseHeaderCell}:before {
+      content: '';
+      position: absolute;
+      border-right: 1px solid ${designTokens.colorNeutral};
+      top: ${designTokens.spacing20};
```

Comment:
It was unclear what the borders top/bottom offset should be.  In the figma file it was 10px, but that is not a spacing value in the design system, so I went with 8px.  This can be changed if necessary.